src/Avalonia.Controls/Platform/ExportAvaloniaModuleAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Avalonia.Platform
+{
+    [AttributeUsage(AttributeTargets.Assembly, AllowMultiple = true)]
+    public class ExportAvaloniaModuleAttribute : Attribute
+    {
+        public ExportAvaloniaModuleAttribute(string name, Type moduleType)
+        {
+            Name = name;
+            ModuleType = moduleType;
+        }
+
+        public string Name { get; private set; }
+        public Type ModuleType { get; private set; }
+
+        public string RequiredWindowingSubsystem { get; set; } = "";
```

Comment:
I think that we need something like "renderer" system from Xamarin.Forms - https://developer.xamarin.com/guides/xamarin-forms/custom-renderer/introduction/
Controls in xforms have an xplat interface and platform-specific implementation that is being looked up by attribute